sbat.c

```diff
@@ -25,51 +25,101 @@ get_sbat_field(CHAR8 *current, CHAR8 *end, const CHAR8 **field, char delim)
 }
 
 EFI_STATUS
-parse_sbat_entry(CHAR8 **current, CHAR8 *end, struct sbat_entry **sbat_entry)
+parse_sbat_section(char *section_base, size_t section_size,
+		   size_t *n_entries,
+		   struct sbat_section_entry ***entriesp)
 {
-	struct sbat_entry *entry = NULL;
+	struct sbat_section_entry *entry = NULL, **entries;
+	EFI_STATUS efi_status = EFI_SUCCESS;
+	list_t csv, *pos = NULL;
+	char * end = section_base + section_size - 1;
+	size_t allocsz = 0;
+	size_t n;
+	char *strtab;
 
-	entry = AllocateZeroPool(sizeof(*entry));
-	if (!entry)
-		return EFI_OUT_OF_RESOURCES;
+	if (!section_base || !section_size || !n_entries || !entriesp)
+		return EFI_INVALID_PARAMETER;
 
-	*current = get_sbat_field(*current, end, &entry->component_name, ',');
-	if (!entry->component_name)
-		goto error;
+	INIT_LIST_HEAD(&csv);
 
-	*current = get_sbat_field(*current, end, &entry->component_generation,
-	                          ',');
-	if (!entry->component_generation)
-		goto error;
+	efi_status = parse_csv_data(section_base, end, 6, &csv);
```

Comment:
Should we have a `#define SBAT_SECTION_COLUMNS 6` or something in `sbat.h` instead of having a magic number here? I know is unlikely that this would change but still would make it easier to read.